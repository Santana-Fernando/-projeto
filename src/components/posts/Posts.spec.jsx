import { render, screen } from "@testing-library/react"
import { Posts } from '.'

const posts = {
    posts: [
        {
            id:1,
            title: 'Title 1',
            body:'Body 01',
            cover: 'img/img.png'
        },
        {
            id:2,
            title: 'Title 2',
            body:'Body 02',
            cover: 'img/img.png'
        },
        {
            id:3,
            title: 'Title 3',
            body:'Body 02',
            cover: 'img/img3.png'
        },
    ]
}

describe('<Posts/>', () => {
    it('should render Posts', () => {
        render(<Posts posts={posts.posts} />)
       
        expect(screen.getAllByRole('heading', {name: /title/i}))
        .toHaveLength(3)

        expect(screen.getAllByRole('img', {name: /title/i}))
        .toHaveLength(3)

        expect(screen.getAllByText(/Body/i))
        .toHaveLength(3)

        expect(screen.getByRole('img', {name: /title 3/i}))
        .toHaveAttribute('src', 'img/img3.png')
    })

    it('should render Posts with empty array', () => {
        render(<Posts />)
       
        expect(screen.queryByRole('heading', {name: /title/i}))
        .not.toBeInTheDocument()
    })

    it('should match snapshot', () => {
        const { container } = render(<Posts posts={posts.posts} />)
       
        expect(container.firstChild).toMatchSnapshot()
        
    })
})