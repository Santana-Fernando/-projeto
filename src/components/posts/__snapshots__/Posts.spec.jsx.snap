// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts/> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="Title 1"
      src="img/img.png"
    />
    <div
      class="post-content"
    >
      <h1>
        1
      </h1>
      <h2>
        Title 1
      </h2>
      <h3>
        Body 01
      </h3>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title 2"
      src="img/img.png"
    />
    <div
      class="post-content"
    >
      <h1>
        2
      </h1>
      <h2>
        Title 2
      </h2>
      <h3>
        Body 02
      </h3>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title 3"
      src="img/img3.png"
    />
    <div
      class="post-content"
    >
      <h1>
        3
      </h1>
      <h2>
        Title 3
      </h2>
      <h3>
        Body 02
      </h3>
    </div>
  </div>
</div>
`;
